src/cpp/src/whisper/whisper.cpp

```diff
@@ -213,7 +212,9 @@ ov::Tensor encode(ov::InferRequest& request,
     raw_metrics.m_inference_durations[0] += MicroSeconds(infer_ms);
 
     // reset input tensor
-    request.set_tensor("input_features", ov::Tensor(ov::element::f32, {0, feature_size, nb_max_frames}));
+    auto m_is_npu = true;
```

Comment:
Need to fix this, how to get here that pipeline is running on NPU?

Reply:
Maybe this reset isn't needed at all? (I don't remember the implementation details)

Reply:
Discussed with @as-suvorov 
Reset is needed here as `request` refers to `mel_data` that can be destroyed later (details in https://github.com/openvinotoolkit/openvino.genai/pull/789#discussion_r1726488275)